prowler/providers/azure/services/monitor/monitor_logging_key_vault_enabled/monitor_logging_key_vault_enabled.py

```diff
@@ -0,0 +1,46 @@
+from prowler.lib.check.models import Check, Check_Report_Azure
+from prowler.providers.azure.services.monitor.monitor_client import monitor_client
+
+
+class monitor_logging_key_vault_enabled(Check):
+    def execute(self) -> Check_Report_Azure:
+        findings = []
+
+        for (
+            subscription_name,
+            diagnostic_settings,
+        ) in monitor_client.diagnostics_settings_for_key_vault.items():
+            report = Check_Report_Azure(self.metadata())
+            if not diagnostic_settings:
+                report.status = "FAIL"
+                report.subscription = subscription_name
+                report.resource_name = "Monitor"
+                report.resource_id = ""
```

Comment:
Resource ID must be filled.

Reply:
This is for the case in which there are no diagnostics in the key vault, so there is no diagnostic_setting.id in this cases,  that's why I left empty. Should I delete resource ID from report in all reports?

Reply:
I have corrected all comments except this one in the new commit